### DISCUSSION TITLE
ask about fallback logic

### DISCUSSION THREAD

**Discussion by wezo3** [savonet/liquidsoap#2458] [General]
Hello Guys,
I want ask about fallback logic in liquidsoap
I have liquidsoap 2.0.4 version and need to do this

relay from remote icecast url as high priority and local track as second priority to local icecast server : 

I use request.dynamic logic in local stream for example the code 
```
def track() =
  uri = list.hd(process.read.lines("/root/loadtracks.py"))
  request.create(uri)
end

localdj=request.dynamic(id="track_input",track)
relay = mksafe(input.http("http://domain.com:8000/stream"))
radio = fallback(track_sensitive = false, [relay, localdj])

output.icecast( %mp3.cbr(stereo=true,stereo_mode="default",samplerate=44100,bitrate=128,id3v2=true),
name="dj",public=true,mount="stream",host="127.0.0.1",port=8002,password="mypass",fallible=true, radio)

```

when i run this relay works good and if relay disconnected i see not switch to the localdj just its empty stream no sound 
I want send relay to local icecast server and when relay disconnected switch to localdj auto at present, and if relay back online again switch auto to relay at present

can any help please about this with liquidsoap ?

Thanks.

**Comment by wezo3** [comment]
@toots .. Can please help us in this hint ?
thank you

**Reply by Russsgithub** [reply]
Hi @wezo3 ,

You need to add localdj to the fallback.

Fallback priorities to the left, so

fallback(track_sensitive=false, [relay, localdj, autodj])

Will play relay if available, localdj if relay drops and autodj if both relay and localdj are down.

The book is a good place to learn how liquidsoap all fits together, and the docs are generally very good.

Have fun.
Russ

**Reply by wezo3** [reply]
thank you for reply, I already in my file use relay and localdj only
autodj i put it by mistake and i updated the topic now
I already use 
radio = fallback(track_sensitive = false, [relay, localdj])
and relay exits start good but if relay stop not switch to localdj just i see stream still online without sound like blank sound
and how switch auto if relay back online and localdj was working i need when relay back online liquidsoap switch auto to relay again at present

**Comment by Russsgithub** [comment]
Ah , I see. Try taking mksafe off relay. This will stream silence if relay is not streaming.

**Reply by wezo3** [reply]
yes that is worked great, thank you so much for your great help, I appreciate
